### PR TITLE
added source code

### PR DESCRIPTION
**Pull request by LittleHydron**

### CODE REVIEW

File: main.c [hunk 1]
```diff
@@ -0,0 +1,68 @@
+#include <stdio.h>
+
+/// Maximum size of the arrays
+const int DIM = 1e5+1;
```
Comment (StasDen) [inline main.c:4]: Why is the number 1e5+1 your restriction?

File: main.c [hunk 2]
```diff
@@ -0,0 +1,68 @@
...
+
+void print_subarray(int arr[], int leftBorder, int rightBorder){
+    while(leftBorder <= rightBorder) printf("%d ", arr[leftBorder ++]);
+    printf("\n");
+    return;
+}
+
+void sort_subarray_in_increasing_order(int arr[], int leftBorder, int rightBorder, int tmp[]){
+/** 
+ *  This function sorts subarray of array <arr> from <leftBorder> to <rightBorder> inclusive using MergeSort algorithm
```
Comment (StasDen) [inline main.c:14]: Why is exactly MergeSort algorithm, not BubbleSort, for instance?

File: main.c [hunk 3]
```diff
@@ -0,0 +1,68 @@
...
+ **/
+    if (leftBorder >= rightBorder) return;
+    int mid = (leftBorder + rightBorder)/2;
+    sort_subarray_in_increasing_order(arr, leftBorder, mid, tmp);
+    sort_subarray_in_increasing_order(arr, mid+1, rightBorder, tmp);
+    int i=leftBorder, j=mid+1, k=leftBorder;
```
Comment (StasDen) [inline main.c:20]: You shuldn't have to use letters such i, j, k(I know it's simpler, but I've got a few troubles while reading)

File: main.c [hunk 4]
```diff
@@ -0,0 +1,68 @@
...
+    }
+    while(leftBorder <= rightBorder){
+        arr[leftBorder] = tmp[leftBorder];
+        leftBorder ++;
+    }
+    return;
+}
+
+int main(){
+    int a[DIM], b[DIM], answer[DIM], tmp[DIM], n, m, k = 0;
```
Comment (StasDen) [inline main.c:43]: What is DIM? https://www.healthline.com/nutrition/dim-supplement

**Review by StasDen — CHANGES_REQUESTED** [review]
Great job! But there is something(to my mind) you should change

File: main.c (hunk 4 above)
Comment (LittleHydron) [reply main.c:43]: DIM is constant. Maximum allowed size of array

File: main.c (hunk 2 above)
Comment (LittleHydron) [reply main.c:14]: MergeSort is one of the fastest sorting algorithms.
BubbleSort is much slower.
MergeSort algorith has O(n*log2(n)) time complexity, while BubbleSort has O(n*n) time complexity.

File: main.c (hunk 1 above)
Comment (LittleHydron) [reply main.c:4]: I don't know
Just big constant

File: main.c (hunk 4 above)
Comment (LittleHydron) [reply main.c:43]: DIM for "Dimension" :)

**Review by LittleHydron — COMMENTED** [review]
Nice work

**Pull request merged**